torch/_inductor/scheduler.py

```diff
@@ -2895,6 +2897,15 @@ def has_shared_data_after_reordering_loop(
 
         return self.score_fusion_memory(node1, node2) > 0
 
+    def unfusable_node(self, node: BaseSchedulerNode) -> bool:
+        """
+        Is this node unfusable under any conditions.
+        """
+        return (
+            isinstance(node, (ExternKernelSchedulerNode, NopKernelSchedulerNode))
+            and not node.is_template()
```

Comment:
Isn't the check for 'not node.is_template()' redundant? Since a template can only be a BaseSchedulerNode or FusedSchedulerNode

Reply:
close to merging, but will address in follow up. there was another place in code base that we check this same set of conditions.. will refactor that as well